### PR TITLE
Remove shell e2e test

### PR DESCRIPTION
**Pull request by sebastienvas**
Removing deprecated e2e test framework in favor of the golang framework

Fixes https://github.com/istio/istio/issues/645
Fixes https://github.com/istio/istio/issues/185
```release-note
none
```

### CODE REVIEW

**Comment by rshriram** [comment]
Please wait. @frankbu FYI.

**Comment by frankbu** [comment]
The shell e2e is the only one that works on minikube, so before it's removed the go version should be fixed to work in minikube too.

**Comment by sebastienvas** [comment]
@frankbu Have you tried the --use-local-cluster when running the tests ?

**Comment by sebastienvas** [comment]
I don't think this is any different:
https://github.com/istio/istio/blob/c2208fe77b23d9f764d9ad38182b3461874ef141/tests/kubeUtils.sh#L84
https://github.com/istio/istio/blob/master/tests/e2e/util/kubeUtils.go#L165

I think the golang framework changes loadbalander to nodePort such that it does not create the ingress.

**Comment by kyessenov** [comment]
@sebastienvas please make sure it runs first, maybe ask someone with minikube (I don't have on installed yet, sorry). The flag name `out-of-cluster` can mean either GKE or minikube, and those have differences between them.

**Comment by sebastienvas** [comment]
@kyessenov I am asking @ZackButcher to test, I am just stating that the shell framework does not have anything different to make it work on minikube.

**Comment by frankbu** [comment]
@sebastienvas I had not tried `--use-local-cluster`, but it does look like it will do the right thing. I think the problem is that you changed the default behavior. With kubeTest.sh, the default was some kind of not-not-out-of-cluster (not sure what that meant), but it worked on minikube without specifying any args. Just running plain `tests/e2e.sh` does not work on minikube. Maybe you can change the default, or at least document clearly how to run on minikube, because I think that plenty of people will be using it.

**Comment by sebastienvas** [comment]
@frankbu I added some documentation. Please review.

File: tests/e2e/README.md [hunk 1]
```diff
@@ -87,6 +87,11 @@ You can build and run this or any single test manually with the same options as
 
 ## Developer process
 
+### Cluster in same local network
+In order to talk to istio ingress, we use the ingress IP by default. If your
+cluster is on the same local network and cannot provide external IP, use the `--use-local-cluster` flag.
+In that case, the framework will not create a LoadBalancer and talk directly to the Pod running istio-ingress.
+
```
Comment (frankbu) [inline tests/e2e/README.md]: This looks good but I would suggest "... cannot provide external IP (for example, minikube), use ...".

I also think you should remove the this sentence from the top of the README: "We recommend GKE cluster although miniKube works for some people too". It works fine for minikube.

**Pull request merged**